.github/workflows/dotnet.yml

```
@@ -17,9 +17,14 @@ jobs:
       uses: actions/setup-dotnet@v1
       with:
         dotnet-version: 6.0.x
-    - name: Restore dependencies
-      run: dotnet restore
-    - name: Build
-      run: dotnet build --no-restore
-    - name: Test
-      run: dotnet test --no-build --verbosity normal
+    - name: Restore packages, build and test
+      run: dotnet test --verbosity normal -c Release -o ./output
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - name: Publish
```

I meant to add this in the previous review but a force push blocked me. 😆 
```suggestion
    steps:
    - name: Setup .NET
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: 6.0.x
    - name: Publish
```

---

I'm not sure how else you'd have access to `nuget` but I might be missing something!

---

🤔 good question I had been tryihg it locally with https://github.com/nektos/act and it was working, my assumption was that the needs will cause that to run on same agent or run the steps again.

If the problem is as you percieve, we'll  need to find a way to cache the output between the jobs too....

I say let's try it this way and fix it it don't work :)

---

Had a quick search, doesn't look like it'll work like it is going to work :S I'll need to update.